### DISCUSSION TITLE
Is it possible to use ecode as a library?

### DISCUSSION THREAD

**Discussion by Sahasrara** [SpartanJ/ecode#112] [Q&A] [answered]
I'm thinking of building a C++ GUI app using Dear IMGUI and am trying to find a text editor I can launch within a window (one that using LSP for code completion).

Is that possible with ecode?

**Comment by SpartanJ** [comment, marked as answer]
Hi! It could be theoretically possible but it's not built for that, so it would require some changes. The base editor is part of the [eepp GUI library](https://github.com/SpartanJ/eepp/) but the LSP is an extension for ecode, so you would need to extend ecode and build it as library, which is not a simple task. There are several text editors available native for IMGUI but I've never seen anyone supporting advanced features like an LSP. ecode is the first big application developed with eepp GUI library, so the only thing I can offer you is to use eepp instead of Dear IMGUI as your GUI app, but eepp is retained mode, so it's a completely different design pattern (it's more similar to Qt).
May be check out [zep](https://github.com/Rezonality/zep), that it's closer to what you need.

**Reply by Sahasrara** [reply]
Thank you very much for your thoughts and help!
